File: platforms/android/src/main/assets/app/broadcastreceivers/NotificationEventReceiver.js
android.support.v4.content.WakefulBroadcastReceiver.extend("com.tns.broadcastreceivers.NotificationEventReceiver", {
    onReceive: function (context, intent) {
        var action = intent.getAction();
        var serviceIntent = null;
        if ("ACTION_START_NOTIFICATION_SERVICE" == action) {
            console.log("onReceive from alarm, starting notification service! thread: " + java.lang.Thread.currentThread().getName());
            serviceIntent = createIntentStartNotificationService(context);
        } else if ("ACTION_DELETE_NOTIFICATION" == action) {
            console.log("onReceive delete notification action, starting notification service to handle delete");
            serviceIntent = createIntentDeleteNotification(context);
        }

        if (serviceIntent) {
            android.support.v4.content.WakefulBroadcastReceiver.startWakefulService(context, serviceIntent);
        }
    }
})

var Intent = android.content.Intent;


function createIntentStartNotificationService(context) {
    var intent = new Intent(context, com.tns.notifications.NotificationIntentService.class);
    intent.setAction("ACTION_START");
    return intent;
}

function createIntentDeleteNotification(context) {
    var intent = new Intent(context, com.tns.notifications.NotificationIntentService.class);
    intent.setAction("ACTION_DELETE");
    return intent;
}

File: platforms/android/src/main/java/com/tns/broadcastreceivers/NotificationEventReceiver.java
package com.tns.broadcastreceivers;

@com.tns.JavaScriptImplementation(javaScriptFile = "./broadcastreceivers/NotificationEventReceiver.js")
public class NotificationEventReceiver extends android.support.v4.content.WakefulBroadcastReceiver implements com.tns.NativeScriptHashCodeProvider {
	public NotificationEventReceiver(){
		super();
		com.tns.Runtime.initInstance(this);
	}

	public void onReceive(android.content.Context param_0, android.content.Intent param_1)  {
		java.lang.Object[] args = new java.lang.Object[2];
		args[0] = param_0;
		args[1] = param_1;
		com.tns.Runtime.callJSMethod(this, "onReceive", void.class, args);
	}

	public boolean equals__super(java.lang.Object other) {
		return super.equals(other);
	}

	public int hashCode__super() {
		return super.hashCode();
	}

}
